### ISSUE TITLE
[docdb] SEGV in libunwind 1.1a on YCQL read from intents

### ISSUE THREAD

**Issue by bmatican** [yugabyte/yugabyte-db#8593]
Cutting this as a separate task from https://github.com/yugabyte/yugabyte-db/issues/6139#issuecomment-844488427, since the YCQL vs YSQL means these might be different issues.

Example stack
```
#0  0x00007f7452af71f5 in access_mem () from /usr/scratch/yugabyte/yugabyte-2.4.0.0-linux/lib/yb-thirdparty/libunwind.so.8
#1  0x00007f7452af7cba in _ULx86_64_step () from /usr/scratch/yugabyte/yugabyte-2.4.0.0-linux/lib/yb-thirdparty/libunwind.so.8
#2  0x00007f7452af68a3 in unw_backtrace () from /usr/scratch/yugabyte/yugabyte-2.4.0.0-linux/lib/yb-thirdparty/libunwind.so.8
#3  0x00007f7447d8bc2b in yb::StackTrace::Collect (this=this@entry=0x7f5fc14d9170, skip_frames=skip_frames@entry=2) at ../../src/yb/util/debug-util.cc:708
#4  0x00007f7447d8bcbd in yb::(anonymous namespace)::HandleStackTraceSignal (signum=<optimized out>) at ../../src/yb/util/debug-util.cc:277
#5  <signal handler called>
#6  0x00007f744f9b4b30 in yb::Slice::compare(yb::Slice const&) const@plt () from /usr/scratch/yugabyte/yugabyte-2.4.0.0-linux/lib/yb/libyb_docdb.so
#7  0x00007f744fa5d0f5 in yb::docdb::IntentAwareIterator::SkipFutureRecords (this=0xbdbc2e400, direction=direction@entry=yb::docdb::Direction::kForward) at ../../src/yb/docdb/intent_aware_iterator.cc:1092
#8  0x00007f744fa60e37 in yb::docdb::IntentAwareIterator::valid (this=this@entry=0xbdbc2e400) at ../../src/yb/docdb/intent_aware_iterator.cc:623
#9  0x00007f744fa1bf89 in yb::docdb::(anonymous namespace)::BuildSubDocument (iter=iter@entry=0xbdbc2e400, data=..., low_ts=..., num_values_observed=num_values_observed@entry=0x7f5fc14daa80,
    this=<optimized out>, this=<optimized out>, this=<optimized out>, this=<optimized out>) at ../../src/yb/docdb/doc_reader.cc:90
#10 0x00007f744fa1c4cf in yb::docdb::(anonymous namespace)::BuildSubDocument (iter=iter@entry=0xbdbc2e400, data=..., low_ts=..., num_values_observed=num_values_observed@entry=0x7f5fc14daa80,
    this=<optimized out>, this=<optimized out>, this=<optimized out>, this=<optimized out>) at ../../src/yb/docdb/doc_reader.cc:218
#11 0x00007f744fa20d2d in yb::docdb::GetSubDocument (db_iter=0xbdbc2e400, data=..., projection=projection@entry=0x0, seek_fwd_suffices=...) at ../../src/yb/docdb/doc_reader.cc:505
#12 0x00007f744fa45d07 in yb::docdb::DocRowwiseIterator::HasNext (this=0xd85460500) at ../../src/yb/docdb/doc_rowwise_iterator.cc:714
#13 0x00007f744f9ea831 in yb::docdb::QLReadOperation::Execute (this=this@entry=0x7f5fc14db730, ql_storage=..., deadline=..., read_time=..., schema=..., projection=..., resultset=0x7f5fc14db6d0,
    restart_read_ht=0x7f5fc14dbd60) at ../../src/yb/docdb/cql_operation.cc:1364
#14 0x00007f74509c4892 in yb::tablet::AbstractTablet::HandleQLReadRequest (this=this@entry=0x1071c0010, deadline=..., read_time=..., ql_read_request=..., txn_op_context=..., result=0x7f5fc14dbcd0)
    at ../../src/yb/tablet/abstract_tablet.cc:54
#15 0x00007f74509e611f in yb::tablet::Tablet::HandleQLReadRequest (this=this@entry=0x1071c0010, deadline=..., read_time=..., ql_read_request=..., transaction_metadata=..., result=0x7f5fc14dbcd0)
    at ../../src/yb/tablet/tablet.cc:1352
#16 0x00007f745133a316 in yb::tserver::TabletServiceImpl::DoReadImpl (this=this@entry=0x44d28c0, read_context=read_context@entry=0x7f5fc14dc290) at ../../src/yb/tserver/tablet_service.cc:2158
#17 0x00007f745133b56c in yb::tserver::TabletServiceImpl::DoRead (this=this@entry=0x44d28c0, read_context=read_context@entry=0x7f5fc14dc290) at ../../src/yb/tserver/tablet_service.cc:2067
#18 0x00007f745133b971 in yb::tserver::TabletServiceImpl::CompleteRead (this=this@entry=0x44d28c0, read_context=read_context@entry=0x7f5fc14dc290) at ../../src/yb/tserver/tablet_service.cc:1973
#19 0x00007f745133d5aa in yb::tserver::TabletServiceImpl::Read (this=this@entry=0x44d28c0, req=req@entry=0x644f1da30, resp=resp@entry=0x107618bf0, context=...) at ../../src/yb/tserver/tablet_service.cc:1964
#20 0x00007f744ddf2d55 in yb::tserver::TabletServerServiceIf::Handle (this=0x44d28c0, call=...) at src/yb/tserver/tserver_service.service.cc:166
#21 0x00007f7449641fd9 in yb::rpc::ServicePoolImpl::Handle (this=0x3b07680, incoming=...) at ../../src/yb/rpc/service_pool.cc:262
#22 0x00007f74495e6fc4 in yb::rpc::InboundCall::InboundCallTask::Run (this=<optimized out>) at ../../src/yb/rpc/inbound_call.cc:212
#23 0x00007f7449653ea8 in yb::rpc::(anonymous namespace)::Worker::Execute (this=<optimized out>) at ../../src/yb/rpc/thread_pool.cc:105
#24 0x00007f7447e763df in std::function<void ()>::operator()() const (this=0x77b7d378)
    at /usr/scratch/yugabyte/yugabyte-2.4.0.0-linux/linuxbrew-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx/Cellar/gcc/5.5.0_4/include/c++/5.5.0/functional:2267
#25 yb::Thread::SuperviseThread (arg=0x77b7d320) at ../../src/yb/util/thread.cc:771
#26 0x00007f744268c694 in start_thread (arg=0x7f5fc14e5700) at pthread_create.c:333
#27 0x00007f7441dc941d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109
```

**Comment by bmatican** [comment]
@spolitov pointed out this is from within libunwind -- we likely wanted to capture a stack as part of the long op tracking mechanism

**Comment by tedyu** [comment]
https://github.com/libunwind/libunwind/issues/94 is fixed in 1.5 release.

Should we use linunwind 1.5 in yugabyte-db-thirdparty ?

**Comment by mbautin** [comment]
@tedyu: +1 for this. Our libunwind version is long overdue for an upgrade.
